src/components/features/home/contact/contact-form/ContactForm.tsx

```diff
@@ -0,0 +1,79 @@
+import { Form } from 'react-router-dom'
+import './contact-form.scss'
+import { useForm } from 'react-hook-form'
+import * as yup from 'yup'
+import { yupResolver } from '@hookform/resolvers/yup'
+
+const schema = yup.object({
+  name: yup
+    .string()
+    .required('Proszę podać swoje imię.')
+    .matches(/^(?=.{1,50}$)[a-z]+(?:['_.\s][a-z]+)*$/i),
+  email: yup
+    .string()
+    .email('Proszę podać poprawny adres email.')
+    .required('Proszę podać swój adres email.'),
+  message: yup.string().required('Proszę podać swoją wiadomość.'),
+  agreeTerms: yup
+    .boolean()
+    .required('Proszę zaakceptować zgodę na przetwarzanie danych osobowych.')
+    .oneOf(
+      [true],
+      'Proszę zaakceptować zgodę na przetwarzanie danych osobowych.'
+    ),
+})
+
+type FormData = yup.InferType<typeof schema>
+
+const ContactForm = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>({
+    resolver: yupResolver(schema),
+  })
+  const onSubmit = (data: FormData) => console.log(data)
+
+  return (
+    <Form className="form" onSubmit={handleSubmit(onSubmit)} method="post">
+      <div className="form__container">
+        <input type="text" placeholder="Imię" {...register('name')} className={errors.name ? 'input__alert-active': ''} />
```

Comment:
Oooo warunkowe klasy! Na moim branch'u wykorzystałam fajną paczkę [classnames](https://www.npmjs.com/package/classnames?activeTab=readme), jeśli chcesz to zachęcam bo duże polecajki są z tą paczką i fajnie się ją używa, przykład ode mnie:

` <p className={classnames('welcome-section__text', clamped && 'clamp')}>`

_(Tutaj "welcome-section__text" jest zawsze a klasa "clamp" jest warunkowa)_ [tutaj więcej na ten temat](https://medium.com/2359media/why-react-classnames-is-a-lifesaver-and-how-you-should-use-it-if-you-arent-already-b25a44ed5603)